utils/common/utils_utf8.h

```diff
@@ -91,6 +91,33 @@ inline std::string wstring_to_utf8(const std::wstring& str)
   return ret;
 }
 
+inline uint8_t utf8_truncate_point(const char* input, size_t length)
+{
+  // Find the beginning of a multibyte char to truncate at and return the
+  // number of bytes to truncate1`
+  if (length < 3)
```

Comment:
I understand this code had existed for some time but I am curious how does this handles 2 byte glyphs, e.g. russian alphabet is encoded in 2 bytes in utf-8.

Reply:
This is a good observation. I believe in this case, the function will return 1 and reduce the byte-length of the input string by one so that the last byte in the input after truncation belongs to a well-formed character.

But I admit that the robustness of this function handling various edge cases for truncation around multi-byte character boundaries needs to be tested.